client/src/pages/Home.js

```diff
@@ -1,29 +1,65 @@
 import MainTable from "../Component/MainTable";
 import SMSModal from "../Component/SMSModal";
 import "./Home.css";
+import React, { useState, useEffect } from "react";
 
 
 
 export function Home() {
-	const [checkedCheckboxes, setCheckedCheckboxes] = useState([]);
+	const [selectedClients, setSelectedClients] = useState([]);
+	const [clientData, setClientData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+
+
+	const loadClientData = () => {
+		setIsLoading(true);
+		fetch("/api/clients")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the customer data, Please try again"
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setClientData(data);
+				setError(null);
+				setSelectedClients([]);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	};
+
+	useEffect(() => {
+		loadClientData();
+	}, []);
 
 	const onHandleSelectedUserState = (isSelected, clientId ) => {
 		if (isSelected) {
-			setCheckedCheckboxes((previousCheckedCheckboxes) => {
-				return previousCheckedCheckboxes.concat(clientId);
+			setSelectedClients((previousSelectedClients) => {
+				return previousSelectedClients.concat(clientId);
 			});
 		} else {
-			setCheckedCheckboxes((previousCheckedCheckboxes) => {
-				return previousCheckedCheckboxes.filter((id) => clientId !== id);
+			setSelectedClients((previousSelectedClients) => {
+				return previousSelectedClients.filter((id) => clientId !== id);
 			});
 		}
 	};
 
-	
+	const onSMSSent = () =>  {
+		loadClientData();
+	};
+
 	return (
 		<main role="main">
-			<MainTable checkedCheckboxes={checkedCheckboxes} onHandleSelectedUserState={onHandleSelectedUserState} />
-			<SMSModal checkedCheckboxes={checkedCheckboxes} />
+			<MainTable selectedClients={selectedClients} onHandleSelectedUserState={onHandleSelectedUserState} clientData={clientData} error={error} isLoading={isLoading} />
+			<SMSModal selectedClients={selectedClients} onSMSSent={onSMSSent} />
```

Comment:
While we're here, what do you think about moving `SMSModal` above `MainTable` so that on the UI the `Send SMS` button is above the table?